File: app/src/main/java/com/shuyu/apprecycler/chat/detail/dagger/component/ChatDetailComponent.java
package com.shuyu.apprecycler.chat.detail.dagger.component;


import com.shuyu.apprecycler.chat.detail.ChatDetailActivity;
import com.shuyu.apprecycler.chat.detail.dagger.ChatDetailSingleton;
import com.shuyu.apprecycler.chat.detail.dagger.module.ChatDetailManagerModule;
import com.shuyu.apprecycler.chat.detail.dagger.module.ChatDetailPresenterModule;
import com.shuyu.apprecycler.chat.detail.dagger.module.ChatDetailViewModule;


import dagger.Component;

@ChatDetailSingleton
@Component(modules = {ChatDetailPresenterModule.class, ChatDetailManagerModule.class, ChatDetailViewModule.class})
public interface ChatDetailComponent {
    void inject(ChatDetailActivity activity);
}

File: app-kotlin/src/main/java/example/shuyu/recycler/kotlin/chat/detail/dagger/component/ChatDetailComponent.kt
package example.shuyu.recycler.kotlin.chat.detail.dagger.component


import dagger.Component
import example.shuyu.recycler.kotlin.chat.detail.ChatDetailActivity
import example.shuyu.recycler.kotlin.chat.detail.dagger.ChatDetailSingleton
import example.shuyu.recycler.kotlin.chat.detail.dagger.module.ChatDetailManagerModule
import example.shuyu.recycler.kotlin.chat.detail.dagger.module.ChatDetailPresenterModule
import example.shuyu.recycler.kotlin.chat.detail.dagger.module.ChatDetailViewModule

@ChatDetailSingleton
@Component(modules = arrayOf(ChatDetailPresenterModule::class, ChatDetailManagerModule::class, ChatDetailViewModule::class))
open interface ChatDetailComponent {
    fun inject(activity: ChatDetailActivity)
}
